[expo-device] Add expo-device module (#4682)
expo/expo: docs/pages/versions/unversioned/sdk/filesystem.md

- **uri (_string_)** -- A `content://` URI pointing to the file. This is the same as the `fileUri` input parameter but in different format.

> ### `FileSystem.getFreeDiskStorageAsync()`

Author: Maybe follow other expo header templates for now since the return types are clearly explained in the `Returns` field.

> ### `FileSystem.getFreeDiskStorageAsync()`

Reviewer: @ninjaachibi @vivianzzhu91 it looks like the error handling behavior in both of these functions differs across platforms -- on Android if an exception is caught, the promise is rejected, whereas on iOS the promise is never rejected but might resolve with nil (undefined). Could you standardize this behavior across platforms and document whichever option you choose? Which one you choose should probably depend on whether you expect the exception/null case to happen in any reasonable scenarios -- if so it might good for the user to expect to handle an occasional `null`/`undefined` value. Otherwise, rejecting the promise would signify that something extra bizarre has happened.

> Gets available internal disk storage size, in bytes. This will get you the free space on the data partition. The data partition hosts all the internal storage for all apps on the device.

Reviewer 2: ```suggestion
Gets the available internal disk storage size, in bytes. This returns the free space on the data partition that hosts all of the internal storage for all apps on the device.
```

#### Example

```javascript
FileSystem.getFreeDiskStorageAsync().then(freeDiskStorage => {
  // Android: 17179869184
  // iOS: 17179869184
});
```

#### Returns

> Returns a Promise that resolves to a number that represents the available internal disk storage size in bytes.

Reviewer 2: ```suggestion
Returns a Promise that resolves to the number of bytes available on the internal disk.
```

### `FileSystem.getTotalDiskCapacityAsync()`

Gets total internal disk storage size, in bytes. This will get you the total capacity on the data partition where hosts all the internal storage for all apps on the device.

#### Example

```javascript
FileSystem.getTotalDiskCapacityAsync().then(totalDiskCapacity => {
  // Android: 17179869184
  // iOS: 17179869184
});
```

#### Returns

> Returns a Promise that resolves to a number that represents the total internal disk storage capacity in bytes.

Reviewer 2: ```suggestion
Returns a Promise that resolves to a number that specifies the total internal disk storage capacity in bytes.
```
